src/com/sun/jna/NativeLibrary.java

```diff
@@ -132,7 +138,7 @@ private static NativeLibrary loadLibrary(String libraryName, Map options) {
 	}
 
         boolean isAbsolutePath = new File(libraryName).isAbsolute();
-        List searchPath = new LinkedList();
+        List<String> searchPath = new LinkedList<String>();
```

Comment:
In case of the referenced line you overlook one point. In line 160, there is a call inserting the custom paths to the beginning of the search path. 
```java
List<String> customPaths = searchPaths.get(libraryName);
if (customPaths != null) {
    synchronized (customPaths) {
        searchPath.addAll(0, customPaths);
    }
}
```
The insert operation for single elements is O(1) for a LinkedList while it is O(n) for an ArrayList. With this bulk insert still all elements of the target ArrayList have to be copied. So I think it is the right choice here from an algorithmic point of view. If you dig deeper and look at how the memory is allocated for this operation, then you have the choice between an array allocation with the memcopy of the elements or you allocate one linked list slot for each added object. It is hard to decide which is better without running some microbenchmarks for the usual number of entries in that use case. The number of entries in this list is usually very low, say 0-10 elements. Which draws me to the conclusion that it probably doesn't matter at all.
Additionally the code you refer to was already there. I only added the Generics here.

However there is also code added by myself it is in the `expandPathsWithSuffix` function. It also uses a LinkedList there. At that part I agree that an ArrayList is faster in case enough memory is allocated when the list is created.

Regarding the point of intention of the chosen structure I would say the latter code should use Collection instead of List. The problem then is that all present structures in the class use List, so replacing that in the function would lead to further changes in the already present code. 

To sum it up I think using Collection in my code may provide a cleaner intent how the data structures should be used. However I do not think it has a performance benefit. On the contrary without further changing the code in `loadLibrary` there would be one more `addAll` call necessary to get a list at the end.
If you insist I can change it accordingly, but I don't think it will have an impact.